beetsplug/web/__init__.py

```diff
@@ -358,9 +359,47 @@ def func(lib, opts, args):
                     r"/*": {"origins": self.config['cors'].get(str)}
                 }
                 CORS(app)
+
+            # Allow serving behind a reverse proxy
+            if self.config['reverse_proxy']:
+                app.wsgi_app = ReverseProxied(app.wsgi_app)
+
             # Start the web application.
             app.run(host=self.config['host'].as_str(),
                     port=self.config['port'].get(int),
                     debug=opts.debug, threaded=True)
         cmd.func = func
         return [cmd]
+
+class ReverseProxied(object):
```

Comment:
This could probably use a link to the source of the code: http://flask.pocoo.org/snippets/35/